framework/widgets/ActiveField.php

```diff
@@ -738,7 +738,9 @@ protected function getClientOptions()
                 $js = $validator->clientValidateAttribute($this->model, $attribute, $this->form->getView());
                 if ($validator->enableClientValidation && $js != '') {
                     if ($validator->whenClient !== null) {
-                        $js = "if (({$validator->whenClient})(attribute, value)) { $js }";
+                        $exp = is_callable($validator->whenClient) ?
```

Comment:
Hmm, the complexity is because we mix PHP with Js here. I can't see how this could be simplified. Actually I also don't think that it's extremely hard to understand, once you got that basic distinction right.